geode-core/src/main/java/org/apache/geode/internal/process/FileProcessController.java

```diff
@@ -112,56 +119,43 @@ private void stop(final File workingDir, final String stopRequestFileName) throw
   private String status(final File workingDir, final String statusRequestFileName,
       final String statusFileName) throws IOException, InterruptedException, TimeoutException {
     // monitor for statusFile
-    final File statusFile = new File(workingDir, statusFileName);
-    final AtomicReference<String> statusRef = new AtomicReference<>();
-
-    final ControlRequestHandler statusHandler = new ControlRequestHandler() {
-      @Override
-      public void handleRequest() throws IOException {
-        // read the statusFile
-        final BufferedReader reader = new BufferedReader(new FileReader(statusFile));
-        final StringBuilder lines = new StringBuilder();
-        try {
-          String line = null;
-          while ((line = reader.readLine()) != null) {
-            lines.append(line);
-          }
-        } finally {
-          statusRef.set(lines.toString());
-          reader.close();
+    File statusFile = new File(workingDir, statusFileName);
+    AtomicReference<String> statusRef = new AtomicReference<>();
+
+    ControlRequestHandler statusHandler = () -> {
```

Comment:
This might be a nice simplification:

```
ControlRequestHandler statusHandler = () -> {
      // read the statusFile
      StringBuilder lines = new StringBuilder();
      try (BufferedReader reader = new BufferedReader(new FileReader(statusFile))) {
        reader.lines().forEach(lines::append);
      } finally {
        statusRef.set(lines.toString());
      }
    };
```